src/common/messageBundles.js

```diff
@@ -0,0 +1,240 @@
+ * @return {String} The name of the distributor grade.
+ */
+gpii.app.messageBundles.getMessageDistributorGrade = function (messageBundles, defaultLocale) {
+    var defaultMessages = messageBundles[defaultLocale],
+        distributions = gpii.app.messageBundles.getMessageDistributions(defaultMessages);
+
+    fluid.defaults("gpii.app.messageDistributor", {
+        gradeNames: "fluid.component",
+        distributeOptions: distributions
+    });
+
+    return "gpii.app.messageDistributor";
+};
+
+/**
+ * Loads synchronously and parses the messageBundles file.
+ * @param messageBundlesPath {String} The path to the messageBundles file relative
+ * to the project directory.
+ * @return {Object} The parsed message bundles for the different locales.
+ */
+gpii.app.messageBundles.loadMessageBundles = function (messageBundlesPath) {
+    var resolvedPath = require("path").resolve(messageBundlesPath);
+    return require(resolvedPath);
+};
+
+/**
+ * Updates the currently used messages depending on the provided locale. In case
+ * there are no messages available for this locale, the default locale messages
+ * will be used.
+ * @param that {Component} The `gpii.app.messageBundles` instance.
+ * @param messageBundles {Object} A map containing the messages for all available
+ * locales.
+ * @param locale {String} The new locale.
+ * @param defaultLocale {String} The default locale.
+ */
+gpii.app.messageBundles.updateMessages = function (that, messageBundles, locale, defaultLocale) {
+    var messages = messageBundles[locale];
+
+    if (!messages) {
+        fluid.log(fluid.logLevel.WARN, "Bundles for locale - " + locale + " - are missing. Using default locale of: " + defaultLocale);
+        messages = messageBundles[defaultLocale];
+    }
+
+    var groupedMessages = gpii.app.messageBundles.groupMessagesByComponent(messages);
+    that.applier.change("messages", groupedMessages);
+};
+
+/**
+ * Given a message key from the `messages` model object, this function returns the
+ * portion of the key which pertains to the name of the component. For example, for
+ * the "gpii_psp_header_autosaveText" key, this function would return "gpii_psp_header".
+ * @param messageKey {String} a key from the `messages` object.
+ * @return {String} The grade name to which this key is related to (except that it will
+ * contain _ insteаd of . as separators).
+ */
+gpii.app.messageBundles.getComponentKey = function (messageKey) {
+    var keyDelimiterIndex = messageKey.lastIndexOf("_");
+    return messageKey.slice(0, keyDelimiterIndex);
+};
+
+/**
+ * Given a message key from the `messages` model object, this function returns the
+ * portion of the key which is the simple message key referenced within the corresponding
+ * component. For example, for the "gpii_psp_header_autosaveText" key, this function would
+ * return "autosaveText".
+ * @param messageKey {String} a key from the `messages` object.
```

Comment:
This docs comment is the wrong way round - we need to import the new linting tasks into gpii-app - could you please update the package.json to include the most up-to-date versions of the linting tasks as seen in https://github.com/GPII/gpii-express/blob/master/package.json#L23 and relint, thanks (note that the version of eslint-config-fluid with the JSDocs linting tasks is 1.3.0, not 1.2.0 as seen here)